### ISSUE TITLE
h5py question/issue

### ISSUE THREAD

**Issue by ghost** [WEC-Sim/WDRT#19]
I ran into an issue when converting the code from Python2 to Python3 - I'm not sure if this is introduced in the Py3 version of h5py, but in the included example h5 file, NDBC46022.h5, there's no date_list value in the file (i.e. instead of including Hs, Tp, date_num, and date_list, there's just Hs, Tp, and date_num), which means none of the h5 I/O or processing functions in ESSC work. Not sure if the file is outdated - could you create a new NDBC46022.h5 file?

**Comment by zmorrell-sand** [comment]
@beseng I had the same issue in the `example_envSampling.py` file.  I went ahead and rebuilt the h5 file in example script by using the data read from the website.  I will push a fixed h5 file, along with an updated `ESSC.py` file that properly reads from the internet when I have permissions for this repo.

@ryancoe Can I have permissions to this repo?

**Comment by rgcoe** [comment]
Yes, done!

--

Ryan Coe, PhD
Water Power Technologies (Org. 8822) | Sandia National Laboratories
Email: Ryan.Coe@sandia.gov<mailto:Ryan.Coe@sandia.gov> | Office: (505)845-9064 | Mobile: (513)910-8809

On Jul 19, 2019, at 3:13 PM, zmorrell-sand <notifications@github.com<mailto:notifications@github.com>> wrote:


@beseng<https://github.com/beseng> I had the same issue in the example_envSampling.py file. I went ahead and rebuilt the h5 file in example script by using the data read from the website. I will push a fixed h5 file, along with an updated ESSC.py file that properly reads from the internet when I have permissions for this repo.

@ryancoe<https://github.com/ryancoe> Can I have permissions to this repo?

**Comment by zmorrell-sand** [comment]
It looks like the h5 file is actually correct on the repo, so I will not update it.  The problem seems to be arising from failure to append to the h5 file in the `example_envSampling` file.  I will not push the commit until this is figured out.

**Comment by zmorrell-sand** [comment]
@ryancoe 
Most of the code conversion seems to be done, but I have two questions regarding the Py2 -> Py3 conversion:

1.  The `example_envSampling.py` file seems to overwrite the `NDBC46022.h5` file.  I am not entirely sure why, but it seems to be dropping the `buoydata` field when it tries to save the contour data.  This is currently the same as the functionality in Python 2, but it seems to be a bug.  Would you like me to try to fix both versions to get what seems to be the intended functionality out?

2.  In the MLER Toolbox, the MLER needs to be extracted from an egg.  That egg seems to have been built with Python 2.7, and I cannot find the source code for this anywhere.  Will we realistically be able to get this converted to Python 3?

Thank you,
Zach

**Comment by zmorrell-sand** [comment]
@ryancoe 
Hi Ryan,

I have figured out how to get the `saveContour` function to not overwrite the buoyData, but I am wanting to make sure that I am implementing intended functionality before moving foward too far.  

I am working under the assumption that when saveContour is called, if no contour data already exists, it should make create new fields for the data.  If there is already existing data, I am having it overwrite the existing contour data.  This is involving quite a few if/else statements, which is causing a decent number of line changes to the code.  It is simple, but time consuming, so I am wanting to make sure that this is the intended functionality before moving forward.

Thanks,
Zach

**Comment by rgcoe** [comment]
@zmorrell-sand - This sounds like the intended functionality. I guess I'm surprised to here that the number of if-then statements is large - shouldn't there just be one to see if there are fields to be overwritten?

**Comment by zmorrell-sand** [comment]
@ryancoe 
The number of if statements is large, because if h5py tries to create a field that already exists, it will crash.  I could have probably just checked to see if one of the fields already exists and it didn't, but if another save function were to save for example the parameters, the function would crash.  I included the if statements for every field to make sure that it couldn't crash.  I already got this done and made sure that it is robust, as I figured it would be a waste of time to just sit waiting, and I could have always reverted my changes.

Also, I was wondering about the MLER Toolbox egg.  Is there source code somewhere for creating this egg? Otherwise, I don't think it runs with python 3.
